packages/passport/sdk/src/confirmation/confirmation.ts

```diff
@@ -149,27 +157,84 @@ export default class ConfirmationScreen {
+        height: popupOptions?.height || CONFIRMATION_WINDOW_HEIGHT,
+      });
+      this.overlay = new Overlay(this.config.popupOverlayOptions);
+    } catch (e) {
+      // If an error is thrown here then the popup is blocked
+      this.overlay = new Overlay(this.config.popupOverlayOptions, true);
+    }
+
+    this.overlay.append(
+      () => {
+        try {
+          this.confirmationWindow?.close();
+          this.confirmationWindow = openPopupCenter({
+            url: this.getHref('loading'),
+            title: CONFIRMATION_WINDOW_TITLE,
+            width: this.popupOptions?.width || CONFIRMATION_WINDOW_WIDTH,
+            height: this.popupOptions?.height || CONFIRMATION_WINDOW_HEIGHT,
+          });
+        } catch { /* Empty */ }
+      },
+      () => this.closeWindow(),
+    );
   }
 
   closeWindow() {
     this.confirmationWindow?.close();
+    this.overlay?.remove();
+    this.overlay = undefined;
   }
 
   showConfirmationScreen(href: string, messageHandler: MessageHandler, resolve: Function) {
-    this.confirmationWindow!.location.href = href;
+    // If popup blocked, the confirmation window will not exist
+    if (this.confirmationWindow) {
+      this.confirmationWindow.location.href = href; // update confirmation window before calling close on old window
+    }
+
+    // This indicates the user closed the overlay so the transaction should be rejected
+    if (!this.overlay) {
+      resolve({ confirmed: false });
+      return;
+    }
+
     // https://stackoverflow.com/questions/9388380/capture-the-close-event-of-popup-window-in-javascript/48240128#48240128
-    const timer = setInterval(() => {
+    const timerCallback = () => {
       if (this.confirmationWindow?.closed) {
-        clearInterval(timer);
+        clearInterval(this.timer);
         window.removeEventListener('message', messageHandler);
         resolve({ confirmed: false });
+        this.confirmationWindow = undefined;
       }
-    }, CONFIRMATION_WINDOW_CLOSED_POLLING_DURATION);
+    };
+    this.timer = setInterval(
+      timerCallback,
+      CONFIRMATION_WINDOW_CLOSED_POLLING_DURATION,
+    );
+    this.overlay.update(() => this.recreateConfirmationWindow(href, timerCallback));
+  }
+
+  private recreateConfirmationWindow(href: string, timerCallback: () => void) {
+    try {
+      // Clears and recreates the timer to ensure when the confirmation window
+      // is closed and recreated the transaction is not rejected.
+      clearInterval(this.timer);
+      this.confirmationWindow?.close();
+      this.confirmationWindow = openPopupCenter({
+        url: href,
+        title: CONFIRMATION_WINDOW_TITLE,
+        width: this.popupOptions?.width || CONFIRMATION_WINDOW_WIDTH,
+        height: this.popupOptions?.height || CONFIRMATION_WINDOW_HEIGHT,
+      });
+      this.timer = setInterval(timerCallback, CONFIRMATION_WINDOW_CLOSED_POLLING_DURATION);
+    } catch { /* Empty */ }
```

Comment:
The purpose of this is to close the popup and reopen but also ensure that the current timer thats firing does not trigger the `closed` if check and rejects the txn.

I tried doing the following but this does not work as expected and the timer still sees that this.confirmationWindow was closed and rejects

```
const oldConfirmationWindow = this.confirmationWindow;
    this.confirmationWindow = openPopupCenter({
      url: href,
      title: CONFIRMATION_WINDOW_TITLE,
      width: this.popupOptions?.width || CONFIRMATION_WINDOW_WIDTH,
      height: this.popupOptions?.height || CONFIRMATION_WINDOW_HEIGHT,
    });
    oldConfirmationWindow?.close();
```

let me know if any concerns with this approach or if there is a better way